### PR TITLE
[Phase 1] LPS-173369 Remove Optional usages for module: portal-search-web-api part2

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:search

File: modules/apps/portal-search/portal-search-web-api/src/main/java/com/liferay/portal/search/web/portlet/shared/search/PortletSharedSearchSettings.java [hunk 1]
```diff
@@ -30,14 +30,10 @@
 @ProviderType
 public interface PortletSharedSearchSettings extends SearchSettings {
 
-	public Optional<String> getParameter71(String name);
-
-	public Optional<String> getParameterOptional(String name);
+	public String getParameter(String name);
```
Comment (dantewang) [inline modules/apps/portal-search/portal-search-web-api/src/main/java/com/liferay/portal/search/web/portlet/shared/search/PortletSharedSearchSettings.java:33]: @julschong Why does this interface still using `Optional` for `getPortletPreferences*`?

Comment (julschong) [reply modules/apps/portal-search/portal-search-web-api/src/main/java/com/liferay/portal/search/web/portlet/shared/search/PortletSharedSearchSettings.java:33]: @dantewang getPortletPreference portion will be on part 3, there is a lot to change for that so will be the final part for this module

**Comment by catarineaguiar** [comment]
ci:stop

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
added baseline

**Comment by Kvale1733** [comment]
ci:test:search

**Comment by catarineaguiar** [comment]
FYI: CI is facing some issues, so this analysis is not so efective.

semver error:
```
     [exec] 
     [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/portal-search/portal-search-web-api] FAILED
     [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
     [exec]     = ================================================== ========== ========== ========== ========== ==========
     [exec]     * com.liferay.portal.search.web.portlet.shared.search MAJOR      5.0.0      5.0.0      6.0.0      VERSION INCREASE REQUIRED
     [exec]     	>   interface  com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchResponse
     [exec]     		>   method     getParameter(java.lang.String,javax.portlet.RenderRequest)
     [exec]     			+   return     java.lang.String
     [exec]     			-   return     java.util.Optional
     [exec]     		>   method     getParameterValues(java.lang.String,javax.portlet.RenderRequest)
     [exec]     			+   return     java.lang.String[]
     [exec]     			-   return     java.util.Optional
     [exec]     	>   interface  com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchSettings
     [exec]     		+   method     getParameter(java.lang.String)
     [exec]     			+   access     abstract
     [exec]     			+   return     java.lang.String
     [exec]     		-   method     getParameter71(java.lang.String)
     [exec]     			-   access     abstract
     [exec]     			-   return     java.util.Optional
     [exec]     		-   method     getParameterOptional(java.lang.String)
     [exec]     			-   access     abstract
     [exec]     			-   return     java.util.Optional
     [exec]     		-   method     getParameterValues71(java.lang.String)
     [exec]     			-   access     abstract
     [exec]     			-   return     java.util.Optional
     [exec]     [Baseline Warning] Bundle Version Change Recommended: 8.0.0
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-03/test-1-38/test-portal-acceptance-pullrequest(master)/6686/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

**Comment by catarineaguiar** [comment]
ci:stop

**Comment by julschong** [comment]
> FYI: CI is facing some issues, so this analysis is not so efective.
> 
> semver error:
> 
> ```
>      [exec] 
>      [exec] com.liferay.semantic.versioning.SemanticVersioningTest > testSemanticVersioning[/apps/portal-search/portal-search-web-api] FAILED
>      [exec]     java.lang.AssertionError:   PACKAGE_NAME                                       DELTA      CUR_VER    BASE_VER   REC_VER    WARNINGS  
>      [exec]     = ================================================== ========== ========== ========== ========== ==========
>      [exec]     * com.liferay.portal.search.web.portlet.shared.search MAJOR      5.0.0      5.0.0      6.0.0      VERSION INCREASE REQUIRED
>      [exec]     	>   interface  com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchResponse
>      [exec]     		>   method     getParameter(java.lang.String,javax.portlet.RenderRequest)
>      [exec]     			+   return     java.lang.String
>      [exec]     			-   return     java.util.Optional
>      [exec]     		>   method     getParameterValues(java.lang.String,javax.portlet.RenderRequest)
>      [exec]     			+   return     java.lang.String[]
>      [exec]     			-   return     java.util.Optional
>      [exec]     	>   interface  com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchSettings
>      [exec]     		+   method     getParameter(java.lang.String)
>      [exec]     			+   access     abstract
>      [exec]     			+   return     java.lang.String
>      [exec]     		-   method     getParameter71(java.lang.String)
>      [exec]     			-   access     abstract
>      [exec]     			-   return     java.util.Optional
>      [exec]     		-   method     getParameterOptional(java.lang.String)
>      [exec]     			-   access     abstract
>      [exec]     			-   return     java.util.Optional
>      [exec]     		-   method     getParameterValues71(java.lang.String)
>      [exec]     			-   access     abstract
>      [exec]     			-   return     java.util.Optional
>      [exec]     [Baseline Warning] Bundle Version Change Recommended: 8.0.0
> ```
> 
> full log [here](https://testray.liferay.com/reports/production/logs/2023-03/test-1-38/test-portal-acceptance-pullrequest(master)/6686/modules-semantic-versioning-jdk8/0/0/jenkins-console.txt.gz?authuser=0)

@catarineaguiar i added baseline here [c57a3a3](https://github.com/liferay-core-infra/liferay-portal/pull/3712/commits/c57a3a3c79eb163d6447bd153e1c6d7c1e732e1e)

could you check to see if this is resolved?

**Comment by catarineaguiar** [comment]
@julschong I will rerun relevant to check if it works better now.

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by Kvale1733** [comment]
@julschong Compile issue found
```
 [exec] > Task :apps:portal-search:portal-search-web:processResources
     [exec] > Task :apps:portal-search:portal-search-web:classes
     [exec] /opt/dev/projects/github/liferay-portal/modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/search/bar/portlet/display/context/factory/SearchBarPortletDisplayContextFactoryTest.java:480: error: no suitable method found for thenReturn(java.util.Optional<java.lang.String>)
     [exec] 		).thenReturn(
     [exec] 		 ^
     [exec]     method org.mockito.stubbing.OngoingStubbing.thenReturn(java.lang.String) is not applicable
     [exec]       (argument mismatch; no instance(s) of type variable(s) T exist so that java.util.Optional<T> conforms to java.lang.String)
     [exec]     method org.mockito.stubbing.OngoingStubbing.thenReturn(java.lang.String,java.lang.String...) is not applicable
     [exec]       (argument mismatch; no instance(s) of type variable(s) T exist so that java.util.Optional<T> conforms to java.lang.String)
     [exec] Note: Some input files use or override a deprecated API.
     [exec] Note: Recompile with -Xlint:deprecation for details.
     [exec] Note: /opt/dev/projects/github/liferay-portal/modules/apps/portal-search/portal-search-web/src/test/java/com/liferay/portal/search/web/internal/facet/display/context/BaseCategoriesSearchFacetDisplayContextTestCase.java uses unchecked or unsafe operations.
     [exec] Note: Recompile with -Xlint:unchecked for details.
     [exec] 1 error
     [exec] 
     [exec] > Task :apps:portal-search:portal-search-web:compileTestJava FAILED
     [exec] 
     [exec] > Task :apps:portal-search:portal-search-web:processTestResources NO-SOURCE
     [exec] > Task :apps:template:template-taglib:autoUpdateXml SKIPPED
```
[Log](https://testray.liferay.com/home/-/testray/case_results/2219192233)

**Pull request closed without merging**